template/build/dev-server.js

```diff
@@ -68,22 +68,34 @@ app.use(staticPath, express.static('./static'))
 var uri = 'http://localhost:' + port
 
 var _resolve
-var readyPromise = new Promise(resolve => {
+var _reject
+var readyPromise = new Promise((resolve, reject) => {
   _resolve = resolve
+  _reject = reject
 })
 
+var server
+var portfinder = require('portfinder')
+portfinder.basePort = port
+
 console.log('> Starting dev server...')
 devMiddleware.waitUntilValid(() => {
-  console.log('> Listening at ' + uri + '\n')
-  // when env is testing, don't need open it
-  if (autoOpenBrowser && process.env.NODE_ENV !== 'testing') {
-    opn(uri)
-  }
-  _resolve()
+  portfinder.getPort((err, port) => {
+    if (err) {
+      _reject(err)
+    }
+    process.env.PORT = port
+    var uri = 'http://localhost:' + port
+    console.log('> Listening at ' + uri + '\n')
+    // when env is testing, don't need open it
+    if (autoOpenBrowser && process.env.NODE_ENV !== 'testing') {
```

Comment:
When we talked about adding a check for 'testing' environment, I didn't think about skipping the `opn(uri)` - I was thinking about skipping the whole portfinder logic.

The point is: for e2e tests, we currently try to access the dev-server through the port that is defined in the config. If the portfinder starts the def-server at a different port because that port is in use, e2e tests will connect to that wrong port anyway.

So either we find a way to tell the e2e test runner about the new found port, or we have to skip portfinder completely hen running e2e tests.

Reply:
I thought we already did that. That is, updating the environment variable read by the e2e config.